float32[] aspeedM
float32[] tspeedM
float32[] tempM
